apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/AppController.java

```diff
@@ -47,6 +52,34 @@ public AppController(
     this.appOpenApiService = appOpenApiService;
   }
 
+  /**
+   * @see com.ctrip.framework.apollo.portal.controller.AppController#create(AppModel)
+   */
+  @Transactional
+  @PreAuthorize(value = "@consumerPermissionValidator.hasCreateApplicationPermission(#request)")
+  @PostMapping(value = "/apps")
+  public void createApp(
+      @RequestBody OpenCreateAppDTO req,
+      HttpServletRequest request
+  ) {
+    if (null == req.getApp()) {
+      throw new BadRequestException("App is null");
+    }
+    final OpenAppDTO app = req.getApp();
+    if (null == app.getAppId()) {
+      throw new BadRequestException("AppId is null");
+    }
+    if (req.isAssignAppRoleToSelf()) {
+      // create app and assign app role to this consumer
+      this.appOpenApiService.createApp(req);
+      long consumerId = this.consumerAuthUtil.retrieveConsumerId(request);
+      consumerService.assignAppRoleToConsumer(consumerId, app.getAppId());
+    } else {
+      // only create app
+      this.appOpenApiService.createApp(req);
+    }
```

Comment:
Would this appear neater?
```suggestion
    // create app
    this.appOpenApiService.createApp(req);
    if (req.isAssignAppRoleToSelf()) {
      long consumerId = this.consumerAuthUtil.retrieveConsumerId(request);
      consumerService.assignAppRoleToConsumer(consumerId, app.getAppId());
    }
```